### PR TITLE
Upgrade the docker infrastructure.

### PR DESCRIPTION
**Pull request by johncarl81**
This includes teh following upgrades:

* Arcgis Runtime to 100.13.0
* Java 17 JDK
* Gradle wrapper
* Maven

Which results in a docker build that runs from scratch.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -6,48 +6,37 @@ ENV NVIDIA_DRIVER_CAPABILITIES \
...
-	echo "oracle-java11-installer-local shared/accepted-oracle-license-v1-2 select true" | debconf-set-selections
+	echo "oracle-java17-installer-local shared/accepted-oracle-license-v1-3 select true" | debconf-set-selections
 
-RUN mkdir -p /var/cache/oracle-jdk11-installer-local/
-COPY docker/jdk-11.0.*_linux-x64_bin.tar.gz /var/cache/oracle-jdk11-installer-local/
+RUN apt-get -qq install ros-melodic-mavros ros-melodic-mavros-extras oracle-java17-installer oracle-java17-set-default openjfx unzip ros-melodic-rosbridge-server
 
-RUN apt-get -qq install ros-melodic-mavros ros-melodic-mavros-extras oracle-java11-installer-local openjfx unzip
+# Install maven
+RUN wget https://downloads.apache.org/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-${MAVEN_VERSION}-bin.tar.gz -P /tmp
```
Comment (Carter90) [inline docker/Dockerfile:24]: For some reason the terminal color got set to red here but I'm not seeing anything obvious that would have done it. It later gets reset back to normal in the next step and it did download.

**Comment by Carter90** [comment]
No luck on my part
```
remove_docker_manual_arcgis ±  ./run.sh 
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details

> Task :run FAILED
No protocol specified
No protocol specified
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.base/sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:1071)
Caused by: java.lang.UnsupportedOperationException: Unable to open DISPLAY
        at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.lambda$new$6(GtkApplication.java:181)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
        at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.<init>(GtkApplication.java:179)
        at javafx.graphics/com.sun.glass.ui.gtk.GtkPlatformFactory.createApplication(GtkPlatformFactory.java:41)
        at javafx.graphics/com.sun.glass.ui.Application.run(Application.java:146)
        at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.startup(QuantumToolkit.java:291)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.startup(PlatformImpl.java:293)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.startup(PlatformImpl.java:163)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.startToolkit(LauncherImpl.java:659)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:410)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:364)
        ... 5 more

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/usr/lib/jvm/java-17-oracle/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 17s
```

I added `--status` to ` /bin/sh -c 'cd /workspace; ./gradlew run --status'` and got
```
No Gradle daemons are running.
   PID STATUS   INFO
    42 STOPPED  (by user or operating system)

Only Daemons for the current Gradle version are displayed. See https://docs.gradle.org/7.3/userguide/gradle_daemon.html#sec:status
```

**Review by Carter90 — APPROVED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (johncarl81) [reply docker/Dockerfile:24]: Im guessing that it's outputting the download status over stderr.  Maybe we could pipe that to stdout or just to /dev/null?

**Comment by Carter90** [comment]
File is no longer online might change it to [downloads.apache.org/maven/maven-3/3.8.5/binaries/apache-maven-3.8.5-bin.tar.gz](downloads.apache.org/maven/maven-3/3.8.5/binaries/apache-maven-3.8.5-bin.tar.gz)
Did a fresh rebuild and got this
```
Step 10/31 : RUN wget https://downloads.apache.org/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-${MAVEN_VERSION}-bin.tar.gz -P /tmp
 ---> Running in eaf3b1d9a1e0
--2022-03-16 00:51:08--  https://downloads.apache.org/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz
Resolving downloads.apache.org (downloads.apache.org)... 88.99.95.219, 135.181.214.104, 2a01:4f9:3a:2c57::2, ...
Connecting to downloads.apache.org (downloads.apache.org)|88.99.95.219|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-03-16 00:51:09 ERROR 404: Not Found.

```

**Comment by Carter90** [comment]
Looks good now had to modify the sim but the dashboard works properly tested manual takeoff, land and RTL on 4 drones, also tested mission takeoff, goto, ddsa and RTL with 4 drones.

**Pull request merged**